File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        totsum=float("-inf")
        cursum=0
        for i in range(len(nums)):
            cursum+=nums[i]
            totsum=max(totsum,cursum)
            if(cursum<0):
                cursum=0
        return totsum
        

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cursum,totsum;
        cursum=0;
        totsum=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            cursum=Math.max(nums[i],cursum+nums[i]);
            totsum=Math.max(totsum,cursum);
        }
        return totsum;
    }
}